### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'paramType')

### ISSUE THREAD

**Issue by mr-cal** [microsoft/pyright#4564]
**Describe the bug**
Pyright 1.1.292 started raising an internal error `TypeError: Cannot read properties of undefined (reading 'paramType')`.

**To Reproduce**
Run `pyright` on the code snippet below.

`pyright<=1.1.291`: :heavy_check_mark: passes
`pyright==1.1.292`:  :x: internal error

**Expected behavior**
No internal errors.

**Screenshots or Code**
Here is a minimal reproducer.  The full source is from the [craft-providers](https://github.com/canonical/craft-providers/blob/f73f983368c725b7dbc6f3c5caafe66f4c97a712/craft_providers/lxd/lxc.py) library.

```python
import enum
import subprocess


class StdinType(enum.Enum):
    INTERACTIVE = subprocess.DEVNULL
    NULL = None


def my_method(stdin:StdinType=StdinType.INTERACTIVE, **kwargs):
    subprocess.run(["ls"], stdin=stdin.value, **kwargs)
```

**VS Code extension or command-line**
Running pyright from `pip` at command line (Ubuntu 22.10 + Python 3.10.7 + pyright 1.1.292).

**Comment by lengau** [comment]
Can confirm this with a fresh pyright 1.1.292 install from yarn on Ubuntu 22.04.

I'm also getting the same error with the version from snap.

**Comment by mr-cal** [comment]
I was able to make an even smaller reproducer without `enum` from a similar failure in the [craft-parts](https://github.com/canonical/craft-parts/pull/359) library:
```
import subprocess

def my_method(cmd, *args, **kwargs):
    return subprocess.run(cmd, *args, **kwargs)
```

Seems specific to passing `**kwargs` to `subprocess`.

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. It's an easy fix, and it will be included in the next release.

**Comment by mr-cal** [comment]
Great, thanks for the speedy triaging @erictraut!

**Comment by erictraut** [comment]
Are you blocked on this, or can you roll back to 1.1.291 for now? I'm considering doing a "hot fix" release of a 1.293 that addresses this particular problem.

**Comment by lengau** [comment]
@erictraut it looks like without a hotfix it'll only be a week? We're *technically* blocked, but if you're planning on releasing 293 next week anyway it's not going to hurt us to wait.

**Comment by lengau** [comment]
Also, thank you for the amazing turnaround time on this!

**Comment by erictraut** [comment]
This is included in pyright 1.1.293, which I just published.
